bccsp/pkcs11/pkcs11.go

```diff
@@ -398,7 +451,7 @@ const (
 	privateKeyType
 )
 
-func findKeyPairFromSKI(mod *pkcs11.Ctx, session pkcs11.SessionHandle, ski []byte, keyType keyType) (*pkcs11.ObjectHandle, error) {
+func findKeyPairFromSKI(mod *pkcs11.Ctx, session pkcs11.SessionHandle, ski []byte, keyType keyType) (pkcs11.ObjectHandle, error) {
```

Comment:
Perhaps break the cache into public and private key handle caches and maintain some of it here instead of at the call site?

Reply:
I'll look into that .... I did think about that option .... need to look and see if there's a specific reason I did not go down that route

Reply:
Decided to stick with a single cache for now